### DISCUSSION TITLE
Not able to stop StreamingResponse once the stream is started

### DISCUSSION THREAD

**Discussion by sheikzaidh** [fastapi/fastapi#8673] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi.applications import FastAPI
import uvicorn
from fastapi.requests import Request
from fastapi.responses import StreamingResponse
from visitor_camera_fastapi import VideoCameraV


app = FastAPI()


async def genv(camera, request):
    while True:
        print(f'is disconnect {request._is_disconnected}')
        frame = camera.get_frame()
        yield (b'--frame\r\n'
            b'Content-Type: image/jpeg\r\n\r\n' + frame + b'\r\n\r\n')

@app.get("/getCameraStream")
async def detectEmployees(request:Request):
    # body = await request.body()
    query = request.query_params
    return StreamingResponse(genv(VideoCameraV(request)), media_type='multipart/x-mixed-replace; boundary=frame')
```

### Description

hi, In my project we capture the stream in the backend and passing to the front-end through API. If the front-end stop getting the stream .then also in the backend stream is getting pass. how Can I stop this stream

### Operating System

Windows

### FastAPI Version

0.63.0

### Python Version

3.7

### Additional Context

I am getting the socket.send() error .there is way I can check the client is disconnect . I have tried with request._is_disconnected also but it returing false if the client disconnected also

**Comment by falkben** [comment, marked as answer]
Try wrapping the code that the streaming response consumes with a try/except and catch `asyncio.CancelledResponse`

This will probably take some experimentation on your part to figure out what works. Should be an async method

But here's some code that catches a basic streaming response data.

```python
import asyncio

import uvicorn
from fastapi import FastAPI
from fastapi.responses import StreamingResponse

app = FastAPI()


# never ending generator for our streaming response
def never_ending_gen():
    while True:
        yield b"some fake data " * 1024  # too small and we never get any output


async def streamer(gen):
    try:
        for i in gen:
            yield i
            await asyncio.sleep(0.25)
    except asyncio.CancelledError:
        print("caught cancelled error")


@app.get("/")
async def root():
    return StreamingResponse(streamer(never_ending_gen()))
```

**Comment by kingofthehill444** [comment]
```
import uvicorn 
from fastapi import FastAPI

# configure the server 
server = uvicorn.Server(config=uvicorn.Config(app=FastAPI()))
)

# Set the force exit flag
server.force_exit = True

# Start the server
server.run()

```

This worked for me. Setting the force exit flag before I run the server.
